calendar-rest-api/src/main/java/com/linagora/calendar/restapi/routes/PeopleSearchRoute.java

```diff
@@ -149,27 +241,54 @@ private UnaryOperator<SearchRequestDTO> validateRequest() {
         };
     }
 
-    public Flux<ResponseDTO> searchPeople(Username username, String query, List<String> objectTypesFilter, int limit) {
-        return Flux.from(contactSearchEngine.autoComplete(AccountId.fromString(username.asString()), query, limit))
-            .flatMap(contact -> getObjectType(contact, objectTypesFilter)
-                .map(objectType -> mapToResponseDTO(objectType).apply(contact)));
+    private Flux<ResponseDTO> search(Username username, String query, List<String> objectTypesFilter, int limit) {
+        return Flux.concat(searchContact(username, query, objectTypesFilter, limit),
+                searchResource(username, query, objectTypesFilter, limit))
+            .take(limit);
+    }
+
+    private Flux<ResponseDTO> searchContact(Username username, String query, List<String> objectTypesFilter, int limit) {
+        if (CollectionUtils.isEmpty(objectTypesFilter) || objectTypesFilter.contains(ObjectType.USER.name().toLowerCase())
+            || objectTypesFilter.contains(ObjectType.CONTACT.name().toLowerCase())) {
+            return Flux.from(contactSearchEngine.autoComplete(AccountId.fromString(username.asString()), query, limit))
+                .flatMap(contact -> resolveUserOrContactType(contact, objectTypesFilter)
+                    .map(lookupResult -> contactToResponseDTO(lookupResult).apply(contact)));
+        }
+        return Flux.empty();
+    }
+
+    private Flux<ResponseDTO> searchResource(Username username, String query, List<String> objectTypesFilter, int limit) {
+        if (CollectionUtils.isEmpty(objectTypesFilter) || objectTypesFilter.contains(ObjectType.RESOURCE.name().toLowerCase())) {
+            return Mono.justOrEmpty(username.getDomainPart())
+                .flatMap(domainDAO::retrieve)
+                .flatMapMany(openPaaSDomain -> resourceDAO.search(openPaaSDomain.id(), query, limit)
+                    .map(resource -> resourceAsResponseDTO(resource, openPaaSDomain.domain())));
+        }
+        return Flux.empty();
+    }
```

Comment:
Maybe we could generify a bit this by adding an interface:

```
interace PeopleSearchProvider {
    Collection<ObjectTypes> supportedObjectTypes();
    Flux<ResponseDTO> search(Username username, String query, List<String> objectTypesFilter, int limit);
}
```

And write a pluggable system that is agnostic to the implementation details of each sub-types.

Thought?